.gitlab/lassen-jobs.yml

```diff
@@ -5,69 +5,75 @@
 # SPDX-License-Identifier: (MIT)
 ##############################################################################
 
-####
-# All lassen build jobs
+# CPU ONLY
+ibm_clang_9:
+  variables:
+    SPEC: "+fortran %clang@9.0.0ibm"
+  extends: .build_and_test_on_lassen
 
-clang_default (build and test on lassen):
+ibm_clang_9_gcc_8:
   variables:
-    SPEC: "%clang@default"
+    SPEC: "+fortran %clang@9.0.0ibm cxxflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 cflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 fflags=-gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1"
   extends: .build_and_test_on_lassen
 
-gcc_default (build and test on lassen):
+gcc_8_3_1:
   variables:
-    SPEC: "%gcc@default"
+    SPEC: "+fortran%gcc@8.3.1"
   extends: .build_and_test_on_lassen
 
-xl_default (build and test on lassen):
+pgi_19_10:
   variables:
-    SPEC: "%xl@default"
+    SPEC: "+fortran%pgi@19.10"
   extends: .build_and_test_on_lassen
 
-pgi_default (build and test on lassen):
+pgi_20_4:
   variables:
-    SPEC: "%pgi@default"
+    SPEC: "+fortran%pgi@20.4"
   extends: .build_and_test_on_lassen
-  allow_failure: true
 
-gcc_8_3_1 (build and test on lassen):
+pgi_20_4_gcc_8_3_1:
   variables:
-    SPEC: "%gcc@8.3.1"
+    SPEC: "+fortran%pgi@20.4 cxxflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 cflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 fflags=-gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1"
```

Comment:
This acts as a reminder for why it can be difficult to share compilers.yaml.
RAJA and Umpire compilers used to be different because there is a combinatorial explosion involving:
- clang and fortran provider
- intel / pgi and gcc toolchain

If we want to be sure projects build with the same toolchain, we need to put things like "gcc-toolchain" directly in the compilers.yaml file. Hopefully, there won’t be too many conflicts to start with.

Note: I _think_ we can adapt the compiler version to reflect the fact we use gcc 8.3.1, cf https://github.com/E4S-Project/e4s-stacks/blob/bbbaf112d6cb74a8253e04a8525518e46ed80332/alcf/iota/compilers.yaml#L29
That way we can have all the configurations we want and share them.

Reply:
That's a reasonable idea. You can see I already messed with the version stuff, e.g. `clang@9.0.0ibm` for example 😄